packages/native-stack/src/views/HeaderConfig.tsx

```diff
@@ -165,8 +165,11 @@ export function HeaderConfig({
    * - If `headerTitle` for Android is specified, so we only need to remove the title and keep the back button
    */
   const backButtonInCustomView = headerBackVisible
-    ? headerLeftElement != null
-    : Platform.OS === 'android' && headerTitleElement != null;
+    ? headerLeftElement != null ||
+      (Platform.OS === 'android' && headerTitleElement != null)
+    : Platform.OS === 'android' &&
+      headerTitleElement != null &&
+      headerLeftElement == null;
```

Comment:
Not gonna lie, this is hard to follow, but I've tested the behaviour & it seems to work nicely.

The `headerBackVisible` is respected (`true` / `false` / `undefined`), presence of `headerLeft` removes back button unless explicitly stated otherwise on both archs. 

Good job :tada: